### ISSUE TITLE
angle brackets in nested lists issue

### ISSUE THREAD

**Issue by harmstyler** [hukkin/mdformat-gfm#47]
Hi,
I have a markdown file with rather strange formatting. The gfm formatter seems to error on it. I reduced it to the broken element in hopes it becomes easier to see. Then angle brackets with a period inside appears to be the cause.

```md

1. foo
   1. foo  <user.name> bar
2. bash
```

**Comment by hukkin** [comment]
Thanks for the issue! I can confirm this happens.

Proper fix would be replacing `linkify` dependency with a `markdown-it-py` plugin that correctly parses GFM autolink extension (can be created in this repository). Linkify matches way more than what's required by the GFM spec, and does it in a peculiar way too, e.g.

```markdown
<!-- This is text only, no link -->

\<user.name>

<!-- This is a link surrounded by angle brackets (brackets are text, not part of the link) -->

<user.name>

<!-- These two are equivalent, just a link, no text brackets-->

<https://python.org>

https://python.org
```

It's not normal that escaping a preceding character that's not part of the (link) syntax itself will invalidate the syntax. There's a chance that `markdown-it` (javascript) may want to change this behavior but not sure.



Anyway, currently links that are matched by `linkify`, but are not CommonMark autolinks, will cause issues to `mdformat-gfm` if they are surrounded by angle brackets.

**Comment by hukkin** [comment]
This is fixed in `mdformat-gfm >=0.4.0`. Instead of `linkify-it-py` we now use a GFM compatible linkifier.

**Comment by hukkin** [comment]
Version `0.4.1` fixes the issue with link and image labels.
